templates/adding_a_new_model/open_model_proposals/ADD_BIG_BIRD.md

```diff
@@ -0,0 +1,1142 @@
+
+Adding a new model is often difficult and requires an in-depth knowledge
+of the 🤗 Transformers library and ideally also of the model\'s original
+repository. At Hugging Face, we are trying to empower the community more
+and more to add models independently. 
+
+The following sections explain in detail how to add BigBird 
+to 🤗 Transformers. You will work closely with Patrick to
+integrate BigBird into 🤗 Transformers. By doing so, you will both gain a 
+theoretical and deep practical understanding of BigBird. 
+But more importantly, you will have made a major
+open-source contribution to 🤗 Transformers. Along the way, you will:
+
+-   get insights into open-source best practices
+-   understand the design principles of one of the most popular NLP
+    libraries
+-   learn how to do efficiently test large NLP models
+-   learn how to integrate Python utilities like `black`, `isort`,
+    `make fix-copies` into a library to always ensure clean and readable
+    code
+
+To start, let\'s try to get a general overview of the Transformers
+library.
+
+General overview of 🤗 Transformers
+----------------------------------
+
+First, you should get a general overview of 🤗 Transformers. 🤗
+Transformers is a very opinionated library, so there is a chance that
+you don\'t agree with some of the library\'s philosophies or design
+choices. From our experience, however, we found that the fundamental
+design choices and philosophies of the library are crucial to
+efficiently scale 🤗 Transformers while keeping maintenance costs at a
+reasonable level.
+
+A good first starting point to better understand the library is to read
+the [documentation of our philosophy](https://huggingface.co/transformers/philosophy.html).
+As a result of our way of working, there are some choices that we try to apply to all models:
+
+-   Composition is generally favored over-abstraction
+-   Duplicating code is not always bad if it strongly improves the
+    readability or accessibility of a model
+-   Model files are as self-contained as possible so that when you read
+    the code of a specific model, you ideally only have to look into the
+    respective `modeling_....py` file.
+
+In our opinion, the library\'s code is not just a means to provide a
+product, *e.g.* the ability to use BERT for inference, but also as the
+very product that we want to improve. Hence, when adding a model, the
+user is not only the person that will use your model, but also everybody
+that will read, try to understand, and possibly tweak your code.
+
+With this in mind, let\'s go a bit deeper into the general library
+design.
+
+### Overview of models
+
+To successfully add a model, it is important to understand the
+interaction between your model and its config,
+`PreTrainedModel`, and `.PretrainedConfig`. For
+exemplary purposes, we will call the model to be added to 🤗 Transformers
+`BrandNewBert`.
+
+Let\'s take a look:
+
+![image](../../../docs/source/imgs/transformers_overview.png)
```

Comment:
This image is fantastic!! Some thoughts:

- I don't know if it's my display or not, but it looks a bit low-resolution, resulting in pixelized output?
- Maybe it should be mentioned what only applies for PyTorch. For example,  in TF you're using a main layer and not the base model itself as a transformer.

Reply:
uploaded a higher resolution one